### ISSUE TITLE
error compiling on powerpc64le with gcc version 5.4.0 

### ISSUE THREAD

**Issue by Shreeshrii** [tesseract-ocr/tesseract#1592]
c++ version
```
 c++ -v
Using built-in specs.
COLLECT_GCC=c++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/powerpc64le-linux-gnu/5/lto-wrapper
Target: powerpc64le-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/IBM 5.4.0-6ubuntu1~16.04.9' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-ppc64el/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-ppc64el --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-ppc64el --with-arch-directory=ppc64le --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-secureplt --with-cpu=power8 --enable-targets=powerpcle-linux --disable-multilib --enable-multiarch --disable-werror --with-long-double-128 --enable-checking=release --build=powerpc64le-linux-gnu --host=powerpc64le-linux-gnu --target=powerpc64le-linux-gnu
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu/IBM 5.4.0-6ubuntu1~16.04.9)

```

commit
```
 git log -3
commit 0a93ad2aaef96023fd863c51c18f110b490a207e
Merge: e9197f3 b4b33dd
Author: zdenop <zdenop@gmail.com>
Date:   Mon May 21 11:26:59 2018 +0200

    Merge pull request #1590 from mduk/patch-1

    Update README.md

commit b4b33dda2a7dda807282781491bd32961bac6ee0
Author: Daniel Kendell <daniel_kendell@macdan.net>
Date:   Mon May 21 09:16:42 2018 +0100

    Update README.md

    Fixed links to C and C++ API header files.

commit e9197f30f3cba6e53e998eb0444e3b33448ba6a2
Merge: 7c8ab18 fedae91
Author: Egor Pugin <egor.pugin@gmail.com>
Date:   Mon May 21 01:28:00 2018 +0300

    Merge pull request #1589 from stweil/fix

    ColPartition: Add missing initialisation for median_left, median_right
```

Error

```
make[1]: Entering directory '/home/ubuntu/tesseract/src/classify'
depbase=`echo featdefs.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../..  -DNDEBUG -I../../src/cutil -I../../src/ccutil -I../../src/ccstruct -I../../src/dict -I../../src/viewer  -I/usr/include/leptonica -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/powerpc64le-linux-gnu/glib-2.0/include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/powerpc64le-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12  -g -O2 -std=c++11 -MT featdefs.lo -MD -MP -MF $depbase.Tpo -c -o featdefs.lo featdefs.cpp &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../.. -DNDEBUG -I../../src/cutil -I../../src/ccutil -I../../src/ccstruct -I../../src/dict -I../../src/viewer -I/usr/include/leptonica -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/powerpc64le-linux-gnu/glib-2.0/include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/powerpc64le-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12 -g -O2 -std=c++11 -MT featdefs.lo -MD -MP -MF .deps/featdefs.Tpo -c featdefs.cpp  -fPIC -DPIC -o .libs/featdefs.o
featdefs.cpp: In function ‘bool ValidCharDescription(const FEATURE_DEFS_STRUCT&, CHAR_DESC)’:
featdefs.cpp:219:36: error: ‘isnan’ was not declared in this scope
           if (isnan(feat->Params[p]) || isinf(feat->Params[p]))
                                    ^
featdefs.cpp:219:36: note: suggested alternative:
In file included from /usr/include/c++/5/random:38:0,
                 from /usr/include/c++/5/bits/stl_algo.h:66,
                 from /usr/include/c++/5/algorithm:62,
                 from ../../src/ccutil/genericvector.h:23,
                 from ../../src/ccstruct/normalis.h:24,
                 from ../../src/ccstruct/blobs.h:33,
                 from ocrfeatures.h:24,
                 from featdefs.h:24,
                 from featdefs.cpp:21:
/usr/include/c++/5/cmath:641:5: note:   ‘std::isnan’
     isnan(_Tp __x)
     ^
featdefs.cpp:219:62: error: ‘isinf’ was not declared in this scope
           if (isnan(feat->Params[p]) || isinf(feat->Params[p]))
                                                              ^
featdefs.cpp:219:62: note: suggested alternative:
In file included from /usr/include/c++/5/random:38:0,
                 from /usr/include/c++/5/bits/stl_algo.h:66,
                 from /usr/include/c++/5/algorithm:62,
                 from ../../src/ccutil/genericvector.h:23,
                 from ../../src/ccstruct/normalis.h:24,
                 from ../../src/ccstruct/blobs.h:33,
                 from ocrfeatures.h:24,
                 from featdefs.h:24,
                 from featdefs.cpp:21:
/usr/include/c++/5/cmath:621:5: note:   ‘std::isinf’
     isinf(_Tp __x)
     ^
Makefile:476: recipe for target 'featdefs.lo' failed
make[1]: *** [featdefs.lo] Error 1
```

**Comment by Shreeshrii** [comment]
Looks like an issue with parenthesis

  if (isnan(feat->Params[p]) || isinf(feat->Params[p]))

**Comment by Shreeshrii** [comment]
Problem seems to be one of the recent commits. I was not able to revert single commits to test, so tried with 2 day old code which compiles.

tesseract -v
tesseract 4.0.0-beta.1-240-g6f4e1
 leptonica-1.75.3
  libgif 5.1.4 : libjpeg 8d (libjpeg-turbo 1.4.2) : libpng 1.2.54 : libtiff 4.0.6 : zlib 1.2.8 : libwebp 0.4.4 : libopenjp2 2.3.0

ubuntu@tesseract-ocr:~/tesseract$ git log -3
commit 6f4e195489c5b6a40a03bd7051770df8a5bfae27
Merge: a9c1233 6c08ec0
Author: zdenop <zdenop@gmail.com>
Date:   Sat May 19 13:17:49 2018 +0200

    Merge pull request #1581 from Shreeshrii/patch-1

    Copy .box and .tif files along with .lstmf files from /tmp

commit a9c123342b3e3059d39e2b13b14aae85767b7625
Merge: 1b8ab8e 663b4d2
Author: Egor Pugin <egor.pugin@gmail.com>
Date:   Sat May 19 13:42:24 2018 +0300

    Merge pull request #1584 from RNCTX/master

    Fix for too many leading slashes before TESSDATA folder

commit 663b4d2d4bd727ddf2ec4885048a1c78cece0162
Author: Robert Clayton <robertnclayton@gmail.com>
Date:   Sat May 19 01:43:31 2018 -0500

    Update Makefile.am

**Comment by AlexanderP** [comment]
The build error on Ubuntu 16.04 under:

- [amd64](https://launchpadlibrarian.net/371267263/buildlog_ubuntu-xenial-amd64.tesseract_4.00~git2479-0a93ad2a-1ppa1~xenial1_BUILDING.txt.gz) 
- [i386 ](https://launchpadlibrarian.net/371267333/buildlog_ubuntu-xenial-i386.tesseract_4.00~git2479-0a93ad2a-1ppa1~xenial1_BUILDING.txt.gz)
- [powerpc](https://launchpad.net/~alex-p/+archive/ubuntu/tesseract-ocr/+build/14917665/+files/buildlog_ubuntu-xenial-ppc64el.tesseract_4.00~git2479-0a93ad2a-1ppa1~xenial1_BUILDING.txt.gz)
- [ppc64el](https://launchpad.net/~alex-p/+archive/ubuntu/tesseract-ocr/+build/14917665/+files/buildlog_ubuntu-xenial-ppc64el.tesseract_4.00~git2479-0a93ad2a-1ppa1~xenial1_BUILDING.txt.gz)
- [s390x](https://launchpad.net/~alex-p/+archive/ubuntu/tesseract-ocr/+build/14917666/+files/buildlog_ubuntu-xenial-s390x.tesseract_4.00~git2479-0a93ad2a-1ppa1~xenial1_BUILDING.txt.gz)

On Ubuntu 14.04 / 17.10 / 18.04 the assembly passes without errors

**Comment by stweil** [comment]
Fixed with pull request #1593.

**Comment by Shreeshrii** [comment]
Thanks!  Builds ok now.

 tesseract -v
tesseract 4.0.0-beta.1-262-g555f
 leptonica-1.75.3
  libgif 5.1.4 : libjpeg 8d (libjpeg-turbo 1.4.2) : libpng 1.2.54 : libtiff 4.0.6 : zlib 1.2.8 : libwebp 0.4.4 : libopenjp2 2.3.0
